### PR TITLE
dev: Make Go directories writeable

### PR DESCRIPTION
**Pull request by olix0r**
`cargo` and `go` may try to write to directories that are part
of the devcontainer's base install. This change ensures that these
directories are writeable by all users in the devcontainer.

This change also updates golangci-lint to latest and adds a
`ReadHeaderTimeout` setting to HTTP servers to prevent slowloris
attacks.

### CODE REVIEW

**Review by mateiidavid — APPROVED** [review]

**Comment by olix0r** [comment]
We should probably make the `ReadHeaderTimeout` change in a separate PR.

**Comment by hawkw** [comment]
Policy integration tests failed due to a [503 while downloading `cargo-action-fmt`](https://github.com/linkerd/linkerd2/runs/7864878020?check_suite_focus=true#step:3:28), I've restarted those.

File: justfile [hunk 1]
```diff
@@ -449,6 +455,13 @@ _devcontainer-build tag target='':
         --target='{{ target }}' \
         --{{ if devcontainer-build-mode == "push" { "push" } else { "load" } }}
 
+##
+## tools
+##
+
+markdownlint:
```
Comment (hawkw) [inline justfile:462]: wondering if this should be `markdown-lint` for consistency with the `go-lint` recipe? OTTOH, `markdownlint` is the name of the tool, so idk...

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -217,49 +217,47 @@ RUN apt update && apt upgrade -y --autoremove \
...
+    && useradd --create-home --uid=1000 --gid=1000 code \
+    && echo "code ALL=(root) NOPASSWD:ALL" >/etc/sudoers.d/code \
+    && chmod 0440 /etc/sudoers.d/code
 
 # Copy node utilities first, because the bin directory include symlinks that we want to preserve.
-COPY --from=node /usr/local/lib/node_modules /usr/local/lib/node_modules
-COPY --from=node /usr/local/include/node /usr/local/include/node
 COPY --from=node /usr/local/man/man1/node.1 /usr/local/man/man1/
+COPY --from=node /usr/local/include/node /usr/local/include/node
+COPY --from=node /usr/local/lib/node_modules /usr/local/lib/node_modules
```
Comment (hawkw) [inline .devcontainer/Dockerfile:231]: is there a reason these were rearranged? not a big deal.

**Review by hawkw — APPROVED** [review]

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (olix0r) [reply .devcontainer/Dockerfile:231]: eh, this is mostly an artifact of changing perms on this (and then undoing it).

File: justfile (hunk 1 above)
Comment (olix0r) [reply justfile:462]: I like this. Simplified to `md-lint`. Added sh-lint, etc.

**Pull request merged**